universal_transfer_operator/docs/guides/concepts.rst

```diff
@@ -0,0 +1,205 @@
+
+    .. literalinclude:: ../../example_dags/example_universal_transfer_operator.py
+       :language: python
+       :start-after: [START dataset_individual_file]
+       :end-before: [END dataset_individual_file]
+
+#. **File pattern or folder location**
+
+    The user can transfer data to and from folder or file pattern. We also resolve the :ref:`file_pattern` in file path based on the :ref:`file_location`. For example:
+
+    .. literalinclude:: ../../example_dags/example_universal_transfer_operator.py
+       :language: python
+       :start-after: [START dataset_file]
+       :end-before: [END dataset_file]
+
+.. _dataframe:
+
+Dataframe
+---------
+
+A DataFrame is another type of Dataset that is becoming more popular in data pipelines. It used to be a temporary processing step within a single task, but now there are more demand and technologies for storing distributed data frames, such as Ray or Snowpark.
+
+A DataFrame Dataset is still a reference object, but its URI may not be clear yet. This is because different frameworks have different ways of representing data frames, such as Spark, Pandas, or Snowpark.
+
+
+.. _api:
+
+API
+---------
+
+An API Dataset refers to data that is obtained or published using a REST API from or to another system. Data pipelines often use this type of Dataset to get data from remote sources in JSON format.
+
+Another type of API Dataset is used to send data to SaaS applications using a REST API. This is sometimes called “Reverse ETL” and it is an emerging operation.
+
+
+.. _transfer_working:
+
+
+How Universal Transfer Operator Works
+-------------------------------------
+.. figure:: /images/approach.png
+
+With universal transfer operator, users can perform data transfers using the following transfer modes:
+
+.. literalinclude:: ../../src/universal_transfer_operator/constants.py
+   :language: python
+   :start-after: [START TransferMode]
+   :end-before: [END TransferMode]
+
+Non-native transfer
+-------------------
+When we load a data located in one dataset located in cloud to another dataset located in cloud, internally the steps involved are:
+
+Steps:
+
+#. Get the dataset data in chunks from dataset storage to the worker node.
+#. Send data to the cloud dataset from the worker node.
+
+This is the default way of transferring datasets. There are performance bottlenecks because of limitations of memory, processing power, and internet bandwidth of worker node.
+
+Following is an example of non-native transfers between Google cloud storage and Sqlite:
+
+.. literalinclude:: ../../example_dags/example_universal_transfer_operator.py
+   :language: python
+   :start-after: [START transfer_non_native_gs_to_sqlite]
+   :end-before: [END transfer_non_native_gs_to_sqlite]
+
+
+Improving bottlenecks by using native transfer
```

Comment:
Should we add an example? What about a link to specific pre-configurations for the users to benefit from them?

Reply:
@tatiana the PR for native transfer is still not merged so an example would be available after that.